pkg/cloud/services/compute/instance.go

```diff
@@ -580,6 +582,21 @@ func (s *Service) InstanceDelete(machine *clusterv1.Machine, openStackMachine *i
 		return err
 	}
 
+	err = util.PollImmediate(RetryIntervalInstanceStatus, TimeoutInstanceDelete, func() (bool, error) {
+		_, err = s.GetInstance(parsed.ID())
+		if err != nil {
+			if capoerrors.IsNotFound(err) {
+				return true, nil
+			}
+			return false, err
+		}
+		return false, nil
+	})
+	if err != nil {
+		record.Warnf(openStackMachine, "FailedDeleteServer", "Failed to deleted server %s: %v", parsed.ID(), err)
```

Comment:
If we want to keep this consistent with the even in deleteBastion we should add the server name here and in l.581